server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java

```diff
@@ -0,0 +1,257 @@
+
+        /**
+         * Create a Map of the results, which is then turned into JSON for logging
+         */
+        public Map<String, Object> toMap() {
+            final Map<String, Object> result = new HashMap<>();
+
+            if (this.indicatorResults != null && this.indicatorResults.size() > 0) {
+                // overall status
+                final HealthStatus status = HealthStatus.merge(this.indicatorResults.stream().map(HealthIndicatorResult::status));
+                result.put("elasticsearch.health.overall.status", status.xContentValue());
+
+                // top-level status for each indicator
+                this.indicatorResults.forEach((indicatorResult) -> {
+                    result.put(
+                        String.format(Locale.ROOT, "%s.%s.status", HEALTH_FIELD_PREFIX, indicatorResult.name()),
+                        indicatorResult.status().xContentValue()
+                    );
+                });
+            }
+
+            return result;
+        }
+    }
+
+    public static final String HEALTH_PERIODIC_LOGGER_POLL_INTERVAL = "health_periodic_logger.poll_interval";
+    public static final Setting<TimeValue> HEALTH_PERIODIC_LOGGER_POLL_INTERVAL_SETTING = Setting.timeSetting(
+        HEALTH_PERIODIC_LOGGER_POLL_INTERVAL,
+        TimeValue.timeValueSeconds(60),
+        TimeValue.timeValueSeconds(15),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    /**
+     * Name constant for the job HealthService schedules
+     */
+    protected static final String HEALTH_PERIODIC_LOGGER_JOB_NAME = "health_periodic_logger";
+
+    private final Settings settings;
+
+    private final ClusterService clusterService;
+    private final Client client;
+
+    private final HealthService healthService;
+    private final Clock clock;
+
+    private final AtomicBoolean isHealthNode = new AtomicBoolean(false);
+    private final SetOnce<SchedulerEngine> scheduler = new SetOnce<>();
+    private volatile TimeValue pollInterval;
+
+    private static final Logger logger = LogManager.getLogger(HealthPeriodicLogger.class);
+
+    /**
+     * Creates a new HealthPeriodicLogger.
+     * This creates a scheduled job using the SchedulerEngine framework and runs it on the current health node.
+     *
+     * @param settings the cluster settings, used to get the interval setting.
+     * @param clusterService the cluster service, used to know when the health node changes.
+     * @param client the client used to call the Health Service.
+     * @param healthService the Health Service, where the actual Health API logic lives.
+     */
+    public HealthPeriodicLogger(Settings settings, ClusterService clusterService, Client client, HealthService healthService) {
+        this.settings = settings;
+        this.clusterService = clusterService;
+        this.client = client;
+        this.healthService = healthService;
+        this.clock = getClock();
```

Comment:
If we want to be able to manipulate the clock during tests, then I would recommend to get it as a constructor argument.

Reply:
It's not used in tests that way, so we can leave it here. I've gotten rid of `getClock()` though.